docs/architecture.md

```diff
@@ -0,0 +1,122 @@
+  - **Advantage**: Maintaining migrations as well as the data access layer code is simplified.
+
+- **Option 2: Use *separate databases***: While using the same database makes sense from a maintenance point of view it might restrict us in the technology choices since a mobile app and backend service do have completely differnt constraints. It might as well hinder us from optimising our backend service as we have to consider mobile constraints.
+  - **Advantage**: Databases can be optimised according to the platform specific needs.
+
+### What kind of database should be used?
+
+- **Option 1: Use a *relational database***: In the past we have used relational databases, promoting a clear data structure and model. A lot of care has been taken to migrate old records to adhere to data model changes. No data redundancy.
+- **Option 2: Use a *no sql database***: A much more flexible option in terms of data model, with performance prioritised over data integrity. Data redundancy.
+
+
+## Lightning node
+
+Our design principles promote self-custody and self-sovereignty - as a result the users lightning node must run on her phone.
+
+### How should the lightning node run on the phone?
+
+- **Option 1: *Foreground process***: The lightning node will be started only when the app is opened.
+  - **Advantages**: low battery consumption, and simple implementation
+- **Option 2: *Background process***: The lightning node runs as background task on the phone, even if the app is closed.
+  - **Advantages**: payments can be received at any time
+
+Another alternative has been discussed if push notifications could be used to notify the user about an incoming payment asking to accept it by opening the app. We have not addressed the question of privacy.
+
+However, this feature might not be important for the MVP.
+
+### How do we extend the lightning channel for DLCs?
+
+DLCs are not supported by the lightning channels outof the box and a custom extension needs to be implemented. Downwards compatibility should be granted though.
+
+ - **Option 1: *Virtual Channels***: ..
+ - **Option 2: ...**:
+
+## State management
+
+How are state changes managed in 10101.
+
+ - **Option 1: *Events***: State changes are reflected with immutable events allowing to recreate a state by applying all events in the correct order. While this architectural pattern introduces some complexities it allows for a much more flexible, async processing. In itchysats an event driven architecture has been implemented for parts of the solution.
+ - **Option 2: *API***: State changes are promoted through synchronous api calls with a blocking wait to check if the call has been processed successfully. This is a simple paradigm an less complex.  
+
+*Note, these options are not exhaustive nor very general for state management. Also the target architecture could make use of a combination of both approaches. Further elaboration and discussion is required here.* 
+
+## Coordinator API
```

Comment:
As far as I can remember, in ItchySats the maker:

- Communicated with the taker via `libp2p`.
- Exposed a HTTP API and a WebSocket API so that a separate component could automate its behaviour.

This section seems to be conflating both aspects, which I found a bit confusing. It might be good to treat them separately.

Reply:
**silly thought**: we could also misuse keysend to exchange custom messages.

While this is definitely a (costly) misuse of the feature, it would mean we stay compatible with all lightning implementations.

Reply:
No need to misuse keysend here. One can send custom messages over the p2p protocol.